project3/coordinator/coordinator.cpp

```diff
@@ -82,19 +99,30 @@ Coordinator::ClientState Coordinator::ReadNextMessage(
 Coordinator::ClientState Coordinator::DispatchMessage(
     const pub_sub_messages::CoordinatorMessage &message) {
   if (message.has_register_()) {
-    LOG_F(INFO, "Handling a register command from Participant %i.",participant_.id);
+    LOG_F(INFO, "Handling a register command from Participant %i.",
+          participant_.id);
     return HandleRequest(message.register_());
   } else if (message.has_deregister()) {
-    LOG_F(INFO, "Handling a deregister command from Participant %i.",participant_.id);
+    DetermineParticipant();
```

Comment:
I'm of the opinion that perhaps these should be called in the actual `HandleRequest` overload where they are needed. Also, since they (theoretically) change with every request, is there any reason anymore to store the participant and `Messenger` in member variables? Maybe you should just make these pure functions that return the relevant instance.

Reply:
Agreed.